handler.go

```diff
@@ -123,17 +123,28 @@ func (fm *Frontman) PostResultsToHub(results []Result) error {
 		}
 	}
 
+	fm.offlineResultsBuffer = append(fm.offlineResultsBuffer, results...)
 	// Bundle hostInfo and results
-	resultsToSend := Results{
+	b, err := json.Marshal(Results{
+		Results: fm.offlineResultsBuffer,
 		HostInfo: hostInfo,
-		Results:  results,
-	}
+	})
 
-	b, err := json.Marshal(resultsToSend)
 	if err != nil {
 		return err
 	}
 
+	// in case buffer + results exceed HubMaxOfflineBufferBytes, reset buffer
+	if len(b) > fm.HubMaxOfflineBufferBytes && len(fm.offlineResultsBuffer) > 0 {
```

Comment:
What's the purpose of the second check `&& len(fm.offlineResultsBuffer)`?
I understand that `if len(b) > fm.HubMaxOfflineBufferBytes` checks if we have more than the defined max amount of bytes to send.
Is it to check if we have to send anything at all?

Reply:
Because no need to reset the empty slice and JSON marshal the same struct again

Reply:
Ok, then I would appreciate a short explanation of that in the comment.
You're handling two cases in one statement but only explain one. This is a bit confusing.

Reply:
ignored because of inactivity